cmd/gossamer/configcmd.go

```diff
@@ -300,20 +296,22 @@ func setRPCConfig(ctx *cli.Context, fig *cfg.RPCCfg) {
 
 }
 
-func startRPC(ctx *cli.Context, fig cfg.RPCCfg, apiSrvc *api.Service) *rpc.HTTPServer {
-	if ctx.GlobalBool(RPCEnabledFlag.Name) {
-		return rpc.NewHTTPServer(apiSrvc.API, &json2.Codec{}, fig.Host, fig.Port, fig.Modules)
+func setupRPC(ctx *cli.Context, fig cfg.RPCCfg, stateSrv *state.Service, networkSrvc *network.Service) *rpc.HTTPServer {
+	cfg := &rpc.HTTPServerConfig{
+		BlockAPI:   stateSrv.Block,
+		StorageAPI: stateSrv.Storage,
+		NetworkAPI: networkSrvc,
```

Comment:
We might need both `stateSrv.Network` and `networkSrvc` in the future. Can we combine the two interfaces before passing to `NetworkAPI`? If not, maybe we should add another property to `HTTPServerConfig` and pass `networkSrvc` to something other than `NetworkAPI` here?

Reply:
not sure I understand, NetworkAPI is an interface and stateSrv.Network and networkSrvc are both structs that implement NetworkAPI. we don't need both right now

Reply:
We don't need it now but I thought we might prepare. Fine as is, just a heads up we might need to make `stateSrv.Netwwork` part of `NetworkAPI` in the future.

Reply:
yeah however I think the RPC only needs those 3 methods from network, so this probably won't need to be updated